apps/web/app/(app)/stats/NewsletterModal.tsx

```diff
@@ -21,27 +21,46 @@ import { SectionHeader } from "@/components/Typography";
 import { EmailList } from "@/components/email-list/EmailList";
 import type { ThreadsResponse } from "@/app/api/google/threads/controller";
 import { Tabs, TabsContent, TabsList, TabsTrigger } from "@/components/ui/tabs";
-import { Button } from "@/components/ui/button";
+import { Button, ButtonLoader } from "@/components/ui/button";
 import { getGmailFilterSettingsUrl } from "@/utils/url";
 import { Tooltip } from "@/components/Tooltip";
 import { AlertBasic } from "@/components/Alert";
 import { onAutoArchive } from "@/utils/actions/client";
-import { MoreDropdown } from "@/app/(app)/bulk-unsubscribe/common";
+import {
+  MoreDropdown,
+  useUnsubscribeButton,
+} from "@/app/(app)/bulk-unsubscribe/common";
 import { useLabels } from "@/hooks/useLabels";
 import { Row } from "@/app/(app)/bulk-unsubscribe/types";
+import { usePremium } from "@/components/PremiumAlert";
+import { usePostHog } from "posthog-js/react";
 
 export function NewsletterModal(props: {
-  newsletter?: Pick<Row, "name" | "lastUnsubscribeLink" | "autoArchived">;
+  newsletter: Pick<Row, "name" | "lastUnsubscribeLink" | "autoArchived">;
   onClose: (isOpen: boolean) => void;
   refreshInterval?: number;
+  mutate?: () => Promise<any>;
```

Comment:
I agree bot. I tried my best to make the suggestion of `mutate?: () => Promise<void>;` work but there were a lot of side effects with it. Would be happy to see a fix for this.

The issue here is that the `mutate` function passed down to this `NewsletterModal` component is to mutate multiple newsletters, but this component only needs to mutate a single newsletter.

The only fixes I could think of was to: 

1. Create a new API route to mutate a single newsletter as well, along with the existing API route to mutate multiple newsletters.
2. Update the `useUnsubscribeButton` hook's logic, and all of its usages.

I think this will need refactoring in the future regardless. Would be happy to have a look at this later.